lib/hooks/create/index.js

```diff
@@ -3,16 +3,26 @@ const { get } = require('lodash');
 module.exports = () => {
   return model => {
     const type = get(model, 'data.model');
-    const action = get(model, 'data.action');
 
     // insta-resolve changes to profiles
     if (type === 'profile' || type === 'trainingModule' || type === 'invitation') {
       return model.setStatus('resolved');
     }
 
     // insta-resolve PIL creation
-    if (type === 'pil' && action === 'create') {
-      return model.setStatus('resolved');
+    if (type === 'pil') {
+      const status = get(model, 'data.modelData.status');
+      const newStatus = get(model, 'data.data.status');
+
+      // inactive PIL updates
+      if (status !== 'active' && newStatus !== 'active') {
+        return model.setStatus('resolved');
+      }
+
+      // deactivating PIL
+      if (status === 'active' && newStatus !== 'active') {
+        return model.setStatus('resolved');
+      }
```

Comment:
I think this is a bit of a long-winded way to say that the action we're trying to undertake is a revocation. I might suggest directly querying the `action` property of the new case data rather than inferring from previous and new state.

i.e. to revoke a PIL send the following payload:

```js
{
  model: 'pil',
  id: 'abc123',
  action: 'revoke'
}
```

And then determine if it's a revocation from that.

Similarly for granting a licence, use an action of `grant` rather than a particular flavour of `update`.

Reply:
👍